### ISSUE TITLE
[Feature]: Easier automation of scrolling by touch

### ISSUE THREAD

**Issue by noname29** [puppeteer/puppeteer#13296]
### Minimal, reproducible example

```TypeScript
I want to imitate a real finger scroll up/down on puppeteer. I use the mobile device configuration on puppeteer.

I don't know which event it is on a browser possibly `wheel` ? so I tried this,
```

### Background

    const puppeteer = require('puppeteer-extra')
    var  {KnownDevices} = require('puppeteer');
    const iPhone = KnownDevices['iPhone 15 Pro'];
    
    async function r()
    {
    	var browserArray = []
    	let browser = await puppeteer.launch(
    		{ 	headless:false,
    			defaultViewport: {
    			  width: 375,
    			  height: 667,
    			  isMobile: true,
    			}/**/
    	});
    
    
    
    	const page = await browser.newPage();
    	await page.emulate(iPhone);
    	async function wheeler()
    	{
    		await page.goto('https://www.google.com/search?q=facebook', {timeout: 0, waitUntil: 'domcontentloaded'});
    
    		await page.mouse.wheel({deltaY: -30});
    
    	}
    	wheeler();
    }
    
    r();

### Expectation

this doesnt do anything, I could use <a href="https://developer.mozilla.org/en-US/docs/Web/API/Element/scrollBy">ScrollBy</a> with puppeteer but I want to imitate a  real finger press scroll event by a human. 

### Reality

As an example, for mouse wheel on browser, I can listen to the event like this: 

    document.body.addEventListener("wheel", (e) => {
      console.log("deltaY: " + e.deltaY)
    });\

How to do the same on mobile browser with finger press scroll up and down ? 

### Puppeteer version

latest

### Node version

latest

### Package manager

npm

### Package manager version

latest

### Operating system

Windows

**Comment by OrKoN** [comment]
Have you tried https://pptr.dev/api/puppeteer.touchscreen / https://github.com/puppeteer/puppeteer/blob/main/test/src/touchscreen.spec.ts#L207? (if you want to test touches and now the mouse). I believe that would trigger the https://developer.mozilla.org/en-US/docs/Web/API/Document/scroll_event event.

**Comment by OrKoN** [comment]
could you clarify the "no effect" part? what effect do you expect?

**Comment by noname29** [comment]
I expect it to scroll down on the webpage, as on mobile you would put your finger and slide it up or down to scroll. I didnt see notice any scroll as I tried with `.touchMove()` @OrKoN

**Comment by OrKoN** [comment]
I see, I think this depends on whether the page implements custom handling of touch events. As you can see Puppeteer will generate custom events for touch here:

```
const puppeteer = require('puppeteer')
const iPhone = puppeteer.KnownDevices['iPhone 15 Pro'];

async function main() {
  let browser = await puppeteer.launch(
    {
      headless: false,
    });

  const page = await browser.newPage();
  await page.emulate(iPhone);

  await page.goto('https://www.google.com/search?q=facebook', { waitUntil: 'networkidle2' });

  await page.evaluate(() => {
    window.addEventListener('touchstart', (e) => {
      console.log('touchstart', e, e.target.id)
    });
    window.addEventListener('touchmove', (e) => {
      console.log('touchmove', e, e.target.id)
    });
  })

  await page.touchscreen.touchStart(150, 300);
  await page.touchscreen.touchMove(150, 310);
  await page.touchscreen.touchMove(150, 320);
  await page.touchscreen.touchMove(150, 330);
  await page.touchscreen.touchMove(150, 340);
  await page.touchscreen.touchMove(150, 350);
  await page.touchscreen.touchMove(150, 360);
  await page.touchscreen.touchEnd();
}

main();
```

That allows you to test your apps but would not trigger the browser scroll behavior. The recommended way is to locate the element you need and use https://pptr.dev/api/puppeteer.elementhandle.scrollintoview to scroll it into view. Or you can evaluate scripts to call the scroll by.

For synthesizing the browser scroll gesture, you could for now try the experimental CDP API:

```
const puppeteer = require('puppeteer')
const iPhone = puppeteer.KnownDevices['iPhone 15 Pro'];

async function main() {
  let browser = await puppeteer.launch(
    {
      headless: false,
    });

  const page = await browser.newPage();
  await page.emulate(iPhone);

  await page.goto('https://www.google.com/search?q=facebook', { waitUntil: 'networkidle2' });

  await page.evaluate(() => {
    window.addEventListener('scroll', (e) => {
      console.log('scroll', e)
    });
  })

  const session = await page.createCDPSession();

  await session.send('Input.synthesizeScrollGesture', {
    x: 150,
    y: 300,
    yDistance: -200,
  });
}

main();
```

**Comment by OrKoN** [comment]
> I believe that would trigger the https://developer.mozilla.org/en-US/docs/Web/API/Document/scroll_event event.

I think I was misremembering about that, it does not appear to be true.

**Comment by noname29** [comment]
Yes It seems `'Input.synthesizeScrollGesture'` is what I am looking for.. I tried 'Input.synthesizeTapGesture' to make a click as if touching on a phone screen, but didn't work ? How can I imitate fingerclick ?

**Comment by OrKoN** [comment]
> Yes It seems `'Input.synthesizeScrollGesture'` is what I am looking for.. I tried 'Input.synthesizeTapGesture' to make a click as if touching on a phone screen, but didn't work ? How can I imitate fingerclick ?

I think for taps, the touchscreen.tap() method should work.

**Comment by noname29** [comment]
Hi,

Any help with scroll using .touchStart() .touchMove() .touchEnd()


CDP fails with many instances, it bugs... I would rather use touch functions

**Comment by noname29** [comment]
```

  await page.touchscreen.touchStart(150, 300);
  await page.touchscreen.touchMove(150, 310);
  await page.touchscreen.touchMove(150, 320);
  await page.touchscreen.touchMove(150, 330);
  await page.touchscreen.touchMove(150, 340);
  await page.touchscreen.touchMove(150, 350);
  await page.touchscreen.touchMove(150, 360);
  await page.touchscreen.touchEnd();
```
  
  
  This actually works very well for scrolling, the only thing is that, I cant adjust speed rate with this, so it scrolls really fast. 

@OrKoN   if you could update these functions to adjust speed rate of touchMove(), it becomes the best
